internal/kubelet/client.go

```diff
@@ -165,9 +166,27 @@ func (p *saClientProvider) BuildClient() (Client, error) {
 	}
 	tr := defaultTransport()
 	tr.TLSClientConfig = &tls.Config{
-		RootCAs: rootCAs,
+		RootCAs:            rootCAs,
+		InsecureSkipVerify: true,
```

Comment:
We have not been passing the user provided `InsecureSkipVerify` config when authType is service account. I am keeping the same behavior here and will create a different issue to address this.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26319 Any ideas/context if this was done intentionally?